### DISCUSSION TITLE
Step by Step instructions for setting up the Lenovo ThinkSmart View

### DISCUSSION THREAD

**Discussion by mngarchow** [dinki/View-Assist#5] [General]
Hi Dinki and all,

I wanted to contribute in a small way. I am not very experienced with Android devices, but I did take all the steps from your videos and write them down to help with inclusion on the wiki. One important piece that might trip people up is the USB aspect. I have a computer that only uses usb3.0 and I could not get QFIL to flash the device. I switched to an older laptop that has usb2.0 and it worked fine. I don't know how to modify the wiki directly, so here they are:

Step by Step (by me) to Flash firmware and setup from Windows 10
Download Files
	1. Download the prerequisite programs to do the flashing: https://bit.ly/3HOIr1g or https://mega.nz/file/EBQ3VSxK#X4DJmfPoFFEAxe8DxIqAIKQershp0Z_dVq-ZXFQ3Vyg
	2. Use this ROM instead of the one mentioned in the first video: https://s3.us-east-1.wasabisys.com/filestash-buk/lenovo-thinksmart-view/combined_kingston_a11_gogapps.7z
	3. (optional) Install 7Zip as needed so you can open the other downloaded files. (https://www.7-zip.org/download.html)

Extract files and install Prerequisites
	1. (optional) Create a directory on your desktop called ThinkSmartView
	2. Extract the downloaded ThinkSmartView.zip to the ThinkSmartView folder
	4. Extract the contents of the file combined_kingston_a11_gogapps.7z to a new folder in the ThinkSmartView folder called combined_kingston_a11_gogapps
	5. Install the Qualcomm Drivers
		1. Double click the file QDLoader HS-USB Driver_64bit_Setup.exe
		2. Yes to the User Account Control prompt
		3. Leave everything as default, click Next (how many times?)
		4. Choose WWAN-DHCP is not used to get IPAddress
		5. Next
		6. Accept license, Next
		7. Install
		8. Finish
		9. Restart PC when prompted at the end.
	6. Install QPST 
		1. Open the QPST 2.7.496 folder
		2. Double click the file QPST.2.7.496.1.exe
		3. Yes to the User Account Control prompt
		4. If prompted, install the VC++ 2013 Redistributable by clicking Install
		5. When prompted, click Next 
		6. Click I accept, then Next
		7. Click Next two more times. Then click Install
		8. Click Finish when done.
	7. Install ABD & Fastboot ++
		1. Double click the file ADB-and-Fastboot++\_v1.0.8.exe
		2. Yes to the User Account Control prompt
		3. Click I accept, then Next
		4. Click Next again.
		5. Leave the checkboxes for desktop shortcuts and Add to System Path options, then click Next.
		6. Click Install.
		7. Uncheck Open the Toolkit and Launch ADB & Fastboot++ options, click Finish.

Flash ROM to device (make sure to use a usb 2.0 port, black not blue, and must use a usb 2.0 hub/bus in windows device manager too)
	1. Open the QFIL tool (click Start then search for QFIL)
		1. Under Select Build Type, pick Flat Build 
		2. Under Select Programmer, browse to the ThinkSmartView\combined_kingston_a11_gogapps folder and select: prog_emmc_firehose_8953_ddr.mbn
		3. Under Select Flat Build, click the Load XML button.
		4. Navigate to the file rawprogram0.xml and click Open
		5. Another file prompt window will open, select patch0.xml and click Open
		6. Unplug the device power cable, plug the usb-c into the computer and the device. (This can be difficult to do for one person, get help as needed)
		7. Hold down both volume buttons while simultaneously plugging in the power. You will hear windows make a sound to indicate it is connected. Let go of volume buttons when you hear the sound.
		8. In QFIL, click on Select Port.
		9. Select the Qualcomm device and click OK. (COM port number will be different for everyone)
		10. Click the Download button and wait while the flasher downloads to the device. It will indicate successful when completed.
		11. Unplug the ThinkSmartView, wait a moment, then plug it back in while holding only the volume up button. 
		12. In the android recovery menu, press the volume up button until you get to Wipe Data/Factory Reset.
		13. Press the volume down button to select Wipe Data/Factory Reset. When prompted "Are you sure?" press volume up to highlight Yes then volume down to select Yes. This will format the data and cache partitions.
		14. When done, it will automatically highlight "Reboot System Now". Press volume down to select and reboot the device. 
		15. The device will boot up with the new ROM installed. A Disney ducks image will display while the device boots. 
		16. The device setup screen will display once boot is completed. Unplug the usb cord. 

Configure
1. Starting on the device setup screen, tap START.
2. Skip
3. Set up wifi
4. Copy Apps and Data, choose Don't Copy.
5. Sign in with your google account and password.
6. On the protect your phone screen, choose Not Now and confirm by choosing Skip Anyway.
7. Choose No Thanks to Google Assistant
8. Choose No Thanks to ???
9. Now on the device home screen.
10. Go to Settings (swipe up) > About Phone > tap on build number at least 7 times to open Developer Mode.
11. Tap Back > System > Advanced > Developer Options > Smallest Width
12. Set Smallest Width to 600
13. Back to settings
14. Settings > Security > Screen Lock > None 
15. Settings > Display > Screen timeout > 30 minutes
16. Done with this section, go back to the home screen.

Install Apps
1. Open the Play Store
2. Search for WebView System Android
3. Pick Android System WebView beta and Install
4. Open the App, Change Provider, choose the 2nd option to allow it to use the new Webview installed instead of the stock application.
5. Press the home button to go back to home.

**Comment by dinki** [comment]
I can't thank you enough for providing this!  I can tell it took a lot of time to create this.  I will move it over to the wiki in the next few days for others to use it.   You are one of the people that make the Home Assistant community great!!

**Reply by mngarchow** [reply]
I'm happy to help more. I'm doing the easy part from my perspective. I haven't started on the View-Assist part yet, but I plan to go through those steps next and try to make improvements. If you make more videos and want me to help with making a script or documenting the steps from your videos just let me know!

**Reply by dinki** [reply]
That would be fantastic and much appreciated.  I'm trying to balance tidying up  the code and making improvements and also moving the documentation forward through video and wiki update.  Any help is much appreciated.

I haven't tried this but maybe you can try editing the wiki and committing your changes and I will review and accept?
https://docs.github.com/en/communities/documenting-your-project-with-wikis/adding-or-editing-wiki-pages#editing-wiki-pages

**Reply by mngarchow** [reply]
I do not have access to the Edit button in step three of the linked instructions. I googled what it should look like and I do not have that. It is probably for maintainers only. I'll read about doing a pull request and see if I can do one of those for the wiki. That might let me edit it and submit it to you for approval without me being a maintainer.

**Comment by pgale** [comment]
Very useful collation of the steps :) Might be worth adding a note about the point where the flash is most likely to fail - at flashing stage if it sits for more than a few seconds without doing anything, this normally indicates that Sahara issue and the flash fails (normally due to USB issues or USB incompatibility).

**Reply by dinki** [reply]
I guess I was lucky and had no issues.  I did see a comment about someone having issues with the driver not being signed under win11.  I had none of these issues so interesting to hear others' experiences and how they over came.  I will be of little use when it comes to technical support, unfortunately, as the goal is really View Assist on a tablet.  This device is great but the ROM still needs some work.

I may get another Thinksmart to do some experimenting of different ROMs once I get the VA stuff out in the open and maybe that will provide a better/easier experience but until then....

**Reply by mngarchow** [reply]
I did include the usb issue I had, which I read about in the other threads linked to the wiki. While I was troubleshooting on my main PC (which uses a Intel USB 3.1 extensible host controller 1.10), I did try various cables without success. I think the cable I used when I moved to my old laptop was a "high quality" cable that came from with my phone, instead of the amazon basics cable I tried on my main computer. The controller on the laptop is called Intel USB 3.0 extensible host controller 1.0. So it may not actually be a 2.0 vs 3.0 issue as much as a 3.0 vs 3.1 issue. I don't have an understanding of how the QFIL tool works well enough to say what exactly the issue is though. I thought all USB connections are supposed to be backwards compatible.

**Reply by pgale** [reply]
No one in any of the many threads I've read on the HA and XDA forums have come up with a good reason. A bit of magic I think. I have a very modern main PC with USB 3.2 gen2 and USB4/thunderbolt and that refused to work on any port. An older PC did initially work for 2-3 flashes and then stopped working and nothing I did got it working again. 
I then went back to a Debian install on a Pi4 and that's been rock solid.

**Comment by mllockwood** [comment]
If you are planning to use more than one of these Lenovo devices, you will need to modify the MAC address of the additional ones on your network.  Instructions here:  https://github.com/dinki/View-Assist/discussions/6

**Reply by dinki** [reply]
Fantastic write up.  I've added it to the wiki with credit to you of course.

https://github.com/dinki/View-Assist/wiki/Lenovo-ThinkSmart#change-mac-address-of-wifi-interface-on-the-lenovo-thinksmart-device-credit-mllockwood

**Comment by pgale** [comment]
If you've read many of my posts (on the HA forums), you'll have seen that I struggled with getting A11 installed and it would just hang at the Lenovo logo. I tried for days with installing most of the stock images (Teams etc) and re-installing A11 over the top. It was very frustrating and I gave up several times as nothing I tried would work.

SUCCESS!

As the A11 ROM misses a few of the .img files I suddenly had a brainwave and built an install with the 'CD-18781Y.201006.099@Teams&Others' ROM with the A11 ROM copied over the top. It WORKED!

Something in my install process just didn't like those missing .img files. This is confirmed on TWO of my devices. I have installed A8.1 on them all, so maybe something has been removed in that process that's needed. Kind of odd that a stock install didn't fix the issue though.

Anyway - I'm happy and off to have another go with A11. Maybe my findings could be written into the wiki in case someone else has a similar issue and has been tinkering with their device in a similar way?

**Comment by mstrblueskys** [comment]
I'm having troubles bootloader unlocking. Do I need to unlock to complete the flash?

**Reply by dinki** [reply]
I wish I could tell you, and maybe others will comment, but I showed all steps I had to take in the videos.  Those videos are 'straight out the box' to flashed.

You may want to check Mattmon's thread on the HA forum for further help.

**Reply by mngarchow** [reply]
> I'm having troubles bootloader unlocking. Do I need to unlock to complete the flash?

I followed the video and made the written instructions from that. I do not remember doing anything related to bootloader unlocking. Is there a specific error message you are getting? I struggled with the process not being able to start the flashing process... but it was something to do with the USB host drivers. I switch to another computer and that worked. I suggest no matter the error, try it on an older computer with a high quality cable to see if that resolves the issue.

**Reply by mstrblueskys** [reply]
Update, I was able to confirm that the bootloader does not need to be unlocked in order to do this. Thanks for the responses!

**Comment by mstrblueskys** [comment]
Has anyone worked on putting the traditional google assistant on this device after the install? I just uninstalled the google go app via these instructions and it sent my device into a boot loop: https://xdaforums.com/t/install-google-app-on-android-go-edition.4383507/

I'm wondering if it's because I had the widget on the home screen.

**Comment by dinki** [comment]
Closing this one as lineage has become the preferred ROM for TSV

**Reply by 1aranzant** [reply]
My wifi keeps disconnecting with Lineage... rendering the device useless :/ 
Anyway, using this technique, I get the following error:
`Download Fail:Sahara Fail:QSaharaServer Fail:Process fail`\
edit: changed USB cable and it then worked
